### PR TITLE
multiCache: Acc + Contract + Code

### PR DESCRIPTION
**Pull request by scf0220**
- Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/okex/okexchain/blob/develop/CONTRIBUTING.md#pr-targeting))

- [ ] Linked to github-issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Wrote tests
- [ ] Updated relevant documentation (`docs/`)
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Reviewed `Files changed` in the github PR explorer

______

For Admin Use:
- Added appropriate labels to PR (ex. wip, ready-for-review, docs)
- Reviewers Assigned
- Squashed all commits, uses message "Merge PR #XYZ: [title]" ([coding standards](https://github.com/okex/coding/blob/master/README.md#merging-a-pr))

### CODE REVIEW

**Comment by scf0220** [comment]
issue:https://github.com/okex/oec/issues/875

File: cmd/exchaind/replay.go [hunk 1]
```diff
@@ -100,6 +100,9 @@ func replayCmd(ctx *server.Context) *cobra.Command {
 	cmd.Flags().Bool(sm.FlagParalleledTx, false, "pall Tx")
 	cmd.Flags().Bool(saveBlock, false, "save block when replay")
 	cmd.Flags().Int64(config.FlagMaxGasUsedPerBlock, -1, "Maximum gas used of transactions in a block")
+	cmd.Flags().Bool(sdk.FlagMultiCache, false, "Enable multi cache")
+	cmd.Flags().Int(sdk.MaxAccInMultiCache, 100000, "max acc in multi cache")
+	cmd.Flags().Int(sdk.MaxContractInMultiCache, 110000, "max contract in multi cache")
```
Comment (scf0220) [inline cmd/exchaind/replay.go]: MaxContract->MaxStroage

File: libs/cosmos-sdk/x/auth/keeper/account.go [hunk 2]
```diff
@@ -26,12 +27,31 @@ func (ak AccountKeeper) NewAccount(ctx sdk.Context, acc exported.Account) export
...
+		ctx.GasMeter().ConsumeGas(gas, "x/auth/keeper/account.go/GetAccount")
+		if data == nil {
+			return nil
+		}
+
+		if isParent {
+			// Avoid overwriting of previous data
+
+			//TODO other better solutions ?
+			return data.Copy().(exported.Account)
```
Comment (scf0220) [inline libs/cosmos-sdk/x/auth/keeper/account.go]: delete isParent

File: libs/cosmos-sdk/types/cache.go [hunk 3]
```diff
@@ -0,0 +1,294 @@
...
+	UseCache = viper.GetBool(FlagMultiCache)
+
+	maxAccInMap = viper.GetInt(MaxAccInMultiCache)
+	deleteAccCount = maxAccInMap / 10
+
+	maxContractInMap = viper.GetInt(MaxContractInMultiCache)
+	deleteContractCount = maxContractInMap / 10
+}
+
+func NewChainCache() *Cache {
```
Comment (scf0220) [inline libs/cosmos-sdk/types/cache.go:81]: add ut

**Review by xiangjianmeng — APPROVED** [review]

**Review by xiangjianmeng — DISMISSED** [review]

**Review by xiangjianmeng — APPROVED** [review]

**Pull request merged**